### PR TITLE
Include pgai extension

### PR DESCRIPTION
**Pull request by jgpruitt**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -190,6 +190,22 @@ RUN set -ex; \
         done; \
     fi
 
+# pgai is an extension for artificial intelligence
+ARG PGAI
```
Comment (graveland) [inline Dockerfile:194]: All the other once have `_VERSION`, so for consistency, please:
```suggestion
ARG PGAI_VERSION
```

**Review by graveland — APPROVED** [review]
I added a couple trivial comments, but otherwise it looks good to me :)

One other comment: you should add a check in `cicd/shared.sh`, probably in the `check_others` function to validate that pgai was actually installed when it was requested.

File: Dockerfile (hunk 1 above)
Comment (jgpruitt) [reply Dockerfile]: fixed

File: Dockerfile [hunk 2]
```diff
@@ -190,6 +190,22 @@ RUN set -ex; \
...
+RUN set -ex; \
+    if [ -n "${PGAI}" ]; then \
+        git clone https://github.com/timescale/pgai.git /build/pgai; \
+        cd /build/pgai; \
+        git checkout "${PGAI}"; \
+        git reset HEAD --hard; \
+        apt-get install -y python3-full python3-pip; \
```
[the remark this answers is not in the document]
Comment (jgpruitt) [reply Dockerfile]: Ok. I separated the python installs and the git/build commands. I checked, and it looks like I won't need python3-full, so I removed it. Also, I'm using a requirements.txt file now, so the python installs need to stay in this section.

File: Dockerfile [hunk 3]
```diff
@@ -190,6 +190,22 @@ RUN set -ex; \
...
+    if [ -n "${PGAI_VERSION}" ]; then \
+        git clone https://github.com/timescale/pgai.git /build/pgai; \
+        cd /build/pgai; \
+        git checkout "${PGAI_VERSION}"; \
+        git reset HEAD --hard; \
+        for pg in ${PG_VERSIONS}; do \
+            cp /build/pgai/ai--*.sql "$(/usr/lib/postgresql/${pg}/bin/pg_config --sharedir)/extension/"; \
+            cp /build/pgai/ai.control "$(/usr/lib/postgresql/${pg}/bin/pg_config --sharedir)/extension/"; \
+        done; \
+        apt-get install -y python3-pip; \
```
Comment (graveland) [inline Dockerfile:205]: If you rebase on master, `python3-pip` is already installed on line 150, so you could drop that.

File: Dockerfile [hunk 4]
```diff
@@ -457,6 +473,9 @@ RUN /build/scripts/install_extensions versions > /.image_config; \
     echo "BUILDER_URL=\"${BUILDER_URL}\"" >> /.image_config; \
     echo "BUILD_DATE=\"$(date -Iseconds)\"" >> /.image_config
 
+#    echo "PGAI_VERSION=\"${PGAI_VERSION}\"" >> /.image_config; \
```
Comment (graveland) [inline Dockerfile:476]: Should uncomment and move this up a few lines, just after PGBACKREST or so?

**Review by graveland — APPROVED** [review]
requirements.txt is a great improvement!

File: Dockerfile (hunk 4 above)
Comment (jgpruitt) [reply Dockerfile:476]: ah yes. thanks

**Pull request merged**